sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -194,12 +195,34 @@ class DataFrame private[sql](
       }
     }
 
-    // Pad the cells
-    rows.map { row =>
-      row.zipWithIndex.map { case (cell, i) =>
-        String.format(s"%-${colWidths(i)}s", cell)
-      }.mkString(" ")
-    }.mkString("\n")
+    // Create SeparateLine
+    val sep:String = {
+      "+" + colWidths.map {
+        size =>
+          val columnSep = new Array[Char](size)
+          for (i <- 0 until size)
+            columnSep(i) = '-'
+          String.valueOf(columnSep)
+      }.mkString("+") + "+\n"
+    }
+
+    sb.append(sep)
+
+    // append column names
+    sb.append("|").append(rows.head.zipWithIndex.map { case (cell, i) =>
+        String.format(s"%${colWidths(i)}s", cell)
```

Comment:
maybe use the padding function on string directly, rather than doing the format.